internal/services/network/web_application_firewall_policy_resource.go

```diff
@@ -801,41 +805,41 @@ func expandWebApplicationFirewallPolicyMatchCondition(input []interface{}) *[]ne
 		matchValues := v["match_values"].([]interface{})
 		transformsRaw := v["transforms"].(*pluginsdk.Set).List()
 
-		var transforms []network.WebApplicationFirewallTransform
+		var transforms []webapplicationfirewallpolicies.WebApplicationFirewallTransform
 		for _, trans := range transformsRaw {
-			transforms = append(transforms, network.WebApplicationFirewallTransform(trans.(string)))
+			transforms = append(transforms, webapplicationfirewallpolicies.WebApplicationFirewallTransform(trans.(string)))
 		}
-		result := network.MatchCondition{
-			MatchValues:      utils.ExpandStringSlice(matchValues),
+		result := webapplicationfirewallpolicies.MatchCondition{
+			MatchValues:      pointer.From(utils.ExpandStringSlice(matchValues)),
 			MatchVariables:   expandWebApplicationFirewallPolicyMatchVariable(matchVariables),
 			NegationConditon: utils.Bool(negationCondition),
-			Operator:         network.WebApplicationFirewallOperator(operator),
+			Operator:         webapplicationfirewallpolicies.WebApplicationFirewallOperator(operator),
 			Transforms:       &transforms,
 		}
 
 		results = append(results, result)
 	}
-	return &results
+	return results
 }
 
-func expandWebApplicationFirewallPolicyMatchVariable(input []interface{}) *[]network.MatchVariable {
-	results := make([]network.MatchVariable, 0)
+func expandWebApplicationFirewallPolicyMatchVariable(input []interface{}) []webapplicationfirewallpolicies.MatchVariable {
```

Comment:
(same here)

Reply:
as above this functioni does not return an error with the slice. so I think a pointer to slice is not needed here, and we can omit a dereference operation in the caller.